factcast-factus/src/test/java/org/factcast/factus/DefaultFactusTest.java

```diff
@@ -297,7 +309,41 @@ void batchAbortedWithException() {
 
     @Nested
     class WhenUpdating {
-        // @Mock private @NonNull managedProjection;
+
+        @Test
+        void updateIsExecutedViaProjection() {
+
+            ManagedProjection m = Mockito.spy(new SimpleProjection());
+            EventApplier<ManagedProjection> ea = Mockito.spy(new DefaultEventApplier<>(mock(
+                    EventSerializer.class), m));
+            when(ehFactory.create(m)).thenReturn(ea);
+            ArgumentCaptor<FactObserver> observer = ArgumentCaptor.forClass(FactObserver.class);
+
+            Fact f1 = Fact.builder().ns("test").type("FooEvent").build("{}");
+            Fact f2 = Fact.builder().ns("test").type("FooEvent").build("{}");
+
+            when(fc.subscribe(any(), observer.capture()))
```

Comment:
I don't think you can use a captor in when, only in verify

Reply:
not true.

`    interface TestFN extends Function<Integer,String>{};

    @Test
    public void testCapturingInBehaviorDefinition(){

        TestFN mock = mock(TestFN.class);
        ArgumentCaptor<Integer> intCap = ArgumentCaptor.forClass(Integer.class);

        when(mock.apply(intCap.capture())).thenReturn("mockedValue");

        String res = mock.apply(112);

        Assertions.assertEquals("mockedValue",res);
        Assertions.assertEquals(112,intCap.getValue());
    }`

Reply:
Oh, indeed, seeing this for the first time... although not recommended :P 

"Warning: it is recommended to use ArgumentCaptor with verification but not with stubbing. Using ArgumentCaptor with stubbing may decrease test readability because captor is created outside of assert (aka verify or 'then') block. Also it may reduce defect localization because if stubbed method was not called then no argument is captured."

Well fine then.